src/super_gradients/training/models/detection_models/yolov5_base.py

```diff
@@ -0,0 +1,503 @@
+"""
+YoloV5 code adapted from https://github.com/ultralytics/yolov5/blob/master/models/yolo.py
+"""
+import math
+from typing import Union, Type, List
+
+import torch
+import torch.nn as nn
+from super_gradients.training.models.detection_models.csp_darknet53 import Conv, CSPDarknet53, get_yolo_version_params
+from super_gradients.training.models.sg_module import SgModule
+from super_gradients.training.utils.detection_utils import non_max_suppression, scale_img, \
+    check_anchor_order, check_img_size_divisibilty, matrix_non_max_suppression, NMS_Type, \
+    DetectionPostPredictionCallback, Anchors
+from super_gradients.training.utils.export_utils import ExportableHardswish, ExportableSiLU
+from super_gradients.training.utils.utils import HpmStruct, get_param
+
+COCO_DETECTION_80_CLASSES_BBOX_ANCHORS = Anchors([[10, 13, 16, 30, 33, 23],
+                                                  [30, 61, 62, 45, 59, 119],
+                                                  [116, 90, 156, 198, 373, 326]],
+                                                 strides=[8, 16, 32])  # output strides of all yolo outputs
+
+ANCHORSLESS_DUMMY_ANCHORS = Anchors([[], [], []], strides=[8, 16, 32])
+
+
+DEFAULT_YOLO_ARCH_PARAMS = {
+    'num_classes': 80,  # Number of classes to predict
+    'depth_mult_factor': 1.0,  # depth multiplier for the entire model
+    'width_mult_factor': 1.0,  # width multiplier for the entire model
+    'channels_in': 3,  # # of classes the model predicts
+    'skip_connections_list': [(12, [6]), (16, [4]), (19, [14]), (22, [10]), (24, [17, 20])],
+    # A list defining skip connections. format is '[target: [source1, source2, ...]]'. Each item defines a skip
+    # connection from all sources to the target according to the layer's index (count starts from the backbone)
+    'connection_layers_input_channel_size': [1024, 1024, 512],
+    # default number off channels for the connecting points between the backbone and the head
+    'fuse_conv_and_bn': False,  # Fuse sequential Conv + B.N layers into a single one
+    'add_nms': False,  # Add the NMS module to the computational graph
+    'nms_conf': 0.25,  # When add_nms is True during NMS predictions with confidence lower than this will be discarded
+    'nms_iou': 0.45,  # When add_nms is True IoU threshold for NMS algorithm
+    # (with smaller value more boxed will be considered "the same" and removed)
+    'yolo_type': 'yoloV5',  # Type of yolo to build: 'yoloV5' and 'yoloX' are supported
+    'yolo_version': 'v6.0',  # Release version of Ultralytics yoloV5 to build a model from: v6.0 and v3.0 are supported
+                             # (has an impact only if yolo_type is yoloV5)
+    'stem_type': None,  # 'focus' and '6x6' are supported, by default is defined by yolo_type and yolo_version
+}
+
+
+class YoloV5PostPredictionCallback(DetectionPostPredictionCallback):
```

Comment:
Check out most of the changed files in #192. There all references to yolov5 are renamed to yolov5_base all over the code. The most sensitive part is the import of `YoloV5PostPredictionCallback` which we have a lot of, including in notebooks. Now it should be imported from yolov5_base

Reply:
Taken care of in https://github.com/Deci-AI/super-gradients/pull/207 . Shouldn't be relevant once we update the branch.